### PR TITLE
fix: remove custom signer seeds from invoke_cpi

### PR DESCRIPTION
**Pull request by ananas-block**
Issue:
- informational suggestion in two audits  to remove custom signer seeds from invoke cpi

Changes:
- remove signer seeds from `InvokeCpiInstructionData`
- enforce default seed `cpi-authority`  instead of custom seeds

### CODE REVIEW

File: merkle-tree/hash-set/src/lib.rs [hunk 1]
```diff
@@ -446,14 +446,16 @@ impl HashSet {
...
                     if first_free_element.is_none() {
                         first_free_element = Some((probe_index, true));
-                        // Since we encountered an empty element we know for sure
-                        // that the element is not in the hash set.
-                        break;
                     }
+                    // Since we encountered an empty bucket we know for sure
+                    // that the element is not in a bucket with higher probe
+                    // index.
+                    break;
```
Comment (SwenSchaeferjohann) [inline merkle-tree/hash-set/src/lib.rs:458]: is this change related to the points in the pr description? @ananas-block

**Review by vadorovsky — APPROVED** [review]
